src/fu-tool.c

```diff
@@ -681,12 +681,12 @@ fu_util_install_blob (FuUtilPrivate *priv, gchar **values, GError **error)
 			  G_CALLBACK (fu_util_update_device_changed_cb), priv);
 
 	/* write bare firmware */
-	if (!fu_engine_install_blob (priv->engine, device,
```

Comment:
Is there a reason we can't just use `fu_engine_install_tasks()` here? Of course, we'd need to create the task here before we do, but it seems a whole lot less work....

Reply:
`fu_engine_install_tasks` looks for and tears apart the CAB to get the blob out.  You won't exactly have a CAB when you're passing to `install-blob`